certora/specs/GovernorFunctions.spec

```diff
@@ -0,0 +1,277 @@
+import "helpers.spec"
+import "Governor.helpers.spec"
+import "GovernorInvariants.spec"
+
+use invariant proposalStateConsistency
+use invariant queuedImplyDeadlineOver
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Rule: propose effect and liveness. Includes "no double proposition"                                                 │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule propose_liveness(uint256 pId, env e) {
+    require nonpayable(e);
+    require clockSanity(e);
+
+    uint8 stateBefore = state(e, pId);
+
+    address[] targets; uint256[] values; bytes[] calldatas; string descr;
+    require pId == hashProposal(targets, values, calldatas, descr);
+    //require validString(descr);
```

Comment:
Can this line be removed?

Reply:
the real answer is: I don't know, this rule timeout so I can't really say much about it.